muon: 0.4.0 -> 0.5.0, unbreak on darwin, enable tests

nekowinston opened the pull request:
Updates muon to 0.5.0 ([changelog](https://docs.muon.build/release_notes.html#-050)). It now includes a meson/muon LSP.

Tests are now passing when building it with embedded samurai. They're run in the package tests, rather than doCheck being enabled by default to keep the required dependencies to a minimum.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nekowinston committed:
muon: 0.4.0 -> 0.5.0, enable tests

nekowinston committed:
muon: unbreak on darwin

isabelroses commented on pkgs/by-name/mu/muon/package.nix:
```diff
@@ -14,29 +16,47 @@
-    (python3.withPackages (ps: [ ps.pyyaml ]))
-    scdoc
-  ];
+  ++ lib.optionals buildDocs [ scdoc ]
+  ++ lib.optionals stdenv.targetPlatform.isDarwin [ apple-sdk ];
```
`apple-sdk` is in darwin stdenv

```suggestion
  ++ lib.optionals buildDocs [ scdoc ];
```

isabelroses reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 435249`
Commit: `c222b842fd9566589712fdd40ce7b41deda53ad2`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
  </ul>
</details>

nekowinston committed:
muon: add updateScript and update meta

SuperSandro2000 commented on pkgs/by-name/mu/muon/package.nix:
```diff
@@ -139,15 +116,39 @@ stdenv.mkDerivation (finalAttrs: {
-    license = licenses.gpl3Only;
-    maintainers = with maintainers; [ ];
-    platforms = platforms.unix;
+  passthru.srcs = {
+    muon-src = fetchFromSourcehut {
+      name = "muon-src";
+      owner = "~lattis";
+      repo = "muon";
+      tag = finalAttrs.version;
+      hash = "sha256-bWEYWUD+GK8R3yVnDTnzFWmm4KAuVPI+1yMfCXWcG/A=";
+    };
+    meson-docs = fetchFromGitHub {
+      name = "meson-docs";
+      repo = "meson-docs";
+      owner = "muon-build";
+      rev = "1017b3413601044fb41ad04977445e68a80e8181";
+      hash = "sha256-aFpyJFIqybLNKhm/kyfCjYylj7DE6muI1+OUh4Cq4WY=";
+    };
+    meson-tests = fetchFromGitHub {
+      name = "meson-tests";
+      repo = "meson-tests";
+      owner = "muon-build";
+      rev = "db92588773a24f67cda2f331b945825ca3a63fa7";
+      hash = "sha256-z4Fc1lr/m2MwIwhXJwoFWpzeNg+udzMxuw5Q/zVvpSM=";
+    };
+  };
+  passthru.updateScript = callPackage ./update.nix { };
```
Why do we need srcs in passthru? srcs should be per default exposed and this creates weird situations with overrides.

nekowinston replied:
`srcs` would be a list, so I would need to use indices to select the source I want in the update script: https://github.com/nekowinston/nixpkgs/blob/49e2204645a00f5e112bce58bfc94e6490afcb63/pkgs/by-name/mu/muon/update.nix#L25
Having it as an attribute set in passhtru allowed for easier access, and I've found some prior art doing the same in nixpkgs: https://github.com/NixOS/nixpkgs/blob/58b0870c75b8535992838d30e02853c67e848cb6/pkgs/by-name/ya/yazi-unwrapped/package.nix#L13

SuperSandro2000 replied:
You can access srcs even when not in passthru.

```
nix-repl> muon.srcs
{
  meson-docs = « derivation /nix/store/wfg6yg8hifx0wz3wy8v95xapnym3dqsj-meson-docs.drv »;
  meson-tests = « derivation /nix/store/2hcxbhcrs6i94g586vlxpccpkp300ss0-meson-tests.drv »;
  muon-src = « derivation /nix/store/9kr8q8afs5izfarcjs8p1q06whxxdivh-muon-src.drv »;
}
```
```diff
diff --git a/pkgs/by-name/mu/muon/package.nix b/pkgs/by-name/mu/muon/package.nix
index fda410e91ff7..545097122ddd 100644
--- a/pkgs/by-name/mu/muon/package.nix
+++ b/pkgs/by-name/mu/muon/package.nix
@@ -20,7 +20,29 @@
   pname = "muon" + lib.optionalString embedSamurai "-embedded-samurai";
   version = "0.5.0";

-  srcs = builtins.attrValues finalAttrs.passthru.srcs;
+  srcs = {
+    muon-src = fetchFromSourcehut {
+      name = "muon-src";
+      owner = "~lattis";
+      repo = "muon";
+      tag = finalAttrs.version;
+      hash = "sha256-bWEYWUD+GK8R3yVnDTnzFWmm4KAuVPI+1yMfCXWcG/A=";
+    };
+    meson-docs = fetchFromGitHub {
+      name = "meson-docs";
+      repo = "meson-docs";
+      owner = "muon-build";
+      rev = "1017b3413601044fb41ad04977445e68a80e8181";
+      hash = "sha256-aFpyJFIqybLNKhm/kyfCjYylj7DE6muI1+OUh4Cq4WY=";
+    };
+    meson-tests = fetchFromGitHub {
+      name = "meson-tests";
+      repo = "meson-tests";
+      owner = "muon-build";
+      rev = "db92588773a24f67cda2f331b945825ca3a63fa7";
+      hash = "sha256-z4Fc1lr/m2MwIwhXJwoFWpzeNg+udzMxuw5Q/zVvpSM=";
+    };
+  };

   sourceRoot = "./muon-src";

@@ -123,29 +145,6 @@
     runHook postInstall
   '';

-  passthru.srcs = {
-    muon-src = fetchFromSourcehut {
-      name = "muon-src";
-      owner = "~lattis";
-      repo = "muon";
-      tag = finalAttrs.version;
-      hash = "sha256-bWEYWUD+GK8R3yVnDTnzFWmm4KAuVPI+1yMfCXWcG/A=";
-    };
-    meson-docs = fetchFromGitHub {
-      name = "meson-docs";
-      repo = "meson-docs";
-      owner = "muon-build";
-      rev = "1017b3413601044fb41ad04977445e68a80e8181";
-      hash = "sha256-aFpyJFIqybLNKhm/kyfCjYylj7DE6muI1+OUh4Cq4WY=";
-    };
-    meson-tests = fetchFromGitHub {
-      name = "meson-tests";
-      repo = "meson-tests";
-      owner = "muon-build";
-      rev = "db92588773a24f67cda2f331b945825ca3a63fa7";
-      hash = "sha256-z4Fc1lr/m2MwIwhXJwoFWpzeNg+udzMxuw5Q/zVvpSM=";
-    };
-  };
   passthru.updateScript = callPackage ./update.nix { };

   meta = {
```

We don't need to make things here more complicated than they need to be.

RossSmyth reviewed (changes requested):
Overall looks fine to me.

Conni2461 commented:
@ofborg eval

SuperSandro2000 commented on pkgs/by-name/mu/muon/update.nix:
```diff
@@ -0,0 +1,35 @@
+    update-source-version "muon" \
+      "$MUON_VERSION" \
+      --version-key=version \
+      --source-key=passthru.srcs.muon-src
+    update-source-version "muon" \
+      "$(curl -s "$REPO/blob/$MUON_VERSION/subprojects/meson-docs.wrap" | grep -oP "revision = \K.+$")" \
+      --version-key=passthru.srcs.meson-docs.rev \
+      --source-key=passthru.srcs.meson-docs
+    update-source-version "muon" \
+      "$(curl -s "$REPO/blob/$MUON_VERSION/subprojects/meson-tests.wrap" | grep -oP "revision = \K.+$")" \
+      --version-key=passthru.srcs.meson-tests.rev \
+      --source-key=passthru.srcs.meson-tests
```
```suggestion
      --source-key=srcs.muon-src
    update-source-version "muon" \
      "$(curl -s "$REPO/blob/$MUON_VERSION/subprojects/meson-docs.wrap" | grep -oP "revision = \K.+$")" \
      --version-key=srcs.meson-docs.rev \
      --source-key=srcs.meson-docs
    update-source-version "muon" \
      "$(curl -s "$REPO/blob/$MUON_VERSION/subprojects/meson-tests.wrap" | grep -oP "revision = \K.+$")" \
      --version-key=srcs.meson-tests.rev \
      --source-key=srcs.meson-tests
```

SuperSandro2000 commented on pkgs/by-name/mu/muon/package.nix:
```diff
@@ -133,17 +123,39 @@ stdenv.mkDerivation (finalAttrs: {
+
+  meta = {
+    homepage = "https://muon.build";
+    description = "An implementation of the meson build system in C99";
```
```suggestion
    description = "Implementation of the meson build system in C99";
```

qweered commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 435249`
Commit: [`b2dd581bd31e5b5370fff8b45423c41d1ce2608b`](https://github.com/NixOS/nixpkgs/commit/b2dd581bd31e5b5370fff8b45423c41d1ce2608b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b2dd581bd31e5b5370fff8b45423c41d1ce2608b..pull/435249/head))
Merge: [`dd055bb6c6cb162e6de3a9dd7eea97cd36eabc9d`](https://github.com/NixOS/nixpkgs/commit/dd055bb6c6cb162e6de3a9dd7eea97cd36eabc9d)

Logs: https://github.com/qweered/nixpkgs-review-gha/actions/runs/17432492669


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>muonStandalone</summary>
<pre>ok     11.20s project+objc+requires_python:objc/1 simple
ok      9.55s project+objc+requires_python:objc/2 nsstring
skip    4.83s project+objc+requires_python:objc/5 objfw
ok      7.53s project+objcpp+requires_python:objcpp/1 simple
ok      0.12s analyze:commands/copyfile.meson
ok      0.12s analyze:commands/coverage.meson
ok      0.16s analyze:commands/delete_suffix.meson
ok      0.12s analyze:commands/i18n/gettext.meson
ok      0.11s analyze:commands/i18n/itstool.meson
ok      0.12s analyze:commands/i18n/msgfmthelper.meson
ok      0.11s analyze:commands/vcs_tagger.meson
ok      0.17s analyze:lib/cmake_prelude.meson
ok      0.18s analyze:modules/_test.meson
ok      8.79s project+objc+requires_python:objc/3 objc args
ok      0.15s analyze:modules/gnome.meson
ok      0.15s analyze:modules/i18n.meson
ok      0.62s fmt:test_all_meson_files
ok      9.15s project+objcpp+requires_python:objcpp/2 objc++ args
skip    6.07s project+objcpp+requires_python:objcpp/3 objfw
ok     12.20s project+objc+requires_python:objc/4 c++ project objc subproject</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>

nekowinston commented:
> ## `nixpkgs-review` result
> 
> [...]
> ### `x86_64-darwin` (sandbox = true)
> ❌ 2 packages failed to build:

Regarding this, I was able to get the tests to build when limiting the test workers arbitrarily with `-j4`. @backwardspy was kind enough to also try this PR on native x86_64-darwin hardware, and without the worker limit, we also hit timeouts there, so it doesn't seem to be just a Rosetta limitation.

I am not sure if we should do an `lib.optionalString (isDarwin && isx86_64) "-j4"` or similar, or just disable doCheck for x86.

nekowinston commented:
I opted for limiting `NIX_BUILD_CORES` for the test phase on `x86_64-darwin`, which seems to work fine on the systems I was able to test, without doing each test 1-by-1.

nekowinston commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 435249`
Commit: [`61175e98cf06277efe91afcabcf549c2798e7aa4`](https://github.com/NixOS/nixpkgs/commit/61175e98cf06277efe91afcabcf549c2798e7aa4) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/61175e98cf06277efe91afcabcf549c2798e7aa4..pull/435249/head))
Merge: [`7338cd65cbcca0918dc068ae0f2dfd1a339fe514`](https://github.com/NixOS/nixpkgs/commit/7338cd65cbcca0918dc068ae0f2dfd1a339fe514)

Logs: https://github.com/nekowinston/nixpkgs-review-gha/actions/runs/17506739578


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>muonStandalone</summary>
<pre>ok      8.52s project+objc+requires_python:objc/1 simple
ok      9.71s project+objc+requires_python:objc/3 objc args
ok     12.57s project+objc+requires_python:objc/2 nsstring
skip    5.08s project+objc+requires_python:objc/5 objfw
ok      8.30s project+objcpp+requires_python:objcpp/1 simple
ok     16.56s project+objc+requires_python:objc/4 c++ project objc subproject
ok      0.21s analyze:commands/copyfile.meson
ok      0.08s analyze:commands/coverage.meson
ok      0.10s analyze:commands/delete_suffix.meson
ok      0.11s analyze:commands/i18n/gettext.meson
ok      0.12s analyze:commands/i18n/itstool.meson
ok      0.38s analyze:commands/i18n/msgfmthelper.meson
ok      0.38s analyze:commands/vcs_tagger.meson
ok      0.11s analyze:lib/cmake_prelude.meson
ok      0.18s analyze:modules/_test.meson
ok      0.18s analyze:modules/gnome.meson
ok      0.16s analyze:modules/i18n.meson
ok      0.89s fmt:test_all_meson_files
ok     10.72s project+objcpp+requires_python:objcpp/2 objc++ args
skip    7.17s project+objcpp+requires_python:objcpp/3 objfw</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>

nekowinston commented:
Ok, I guess GitHub Actions runners on x86 are still too slow. I have no idea if the same applies to ofborg, but I'll just turn off parallel builds for x86_64-darwin altogether.

nekowinston commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 435249`
Commit: [`51f499be9cff6cae8ae9246f930e2b449d1c940c`](https://github.com/NixOS/nixpkgs/commit/51f499be9cff6cae8ae9246f930e2b449d1c940c) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/51f499be9cff6cae8ae9246f930e2b449d1c940c..pull/435249/head))
Merge: [`2c217088d2930f2b248ed70da8bea6e2b5cbe785`](https://github.com/NixOS/nixpkgs/commit/2c217088d2930f2b248ed70da8bea6e2b5cbe785)

Logs: https://github.com/nekowinston/nixpkgs-review-gha/actions/runs/17507205655


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>muonStandalone</summary>
<pre>ok      5.24s project+objc+requires_python:objc/1 simple
ok      7.13s project+objc+requires_python:objc/2 nsstring
ok      5.09s project+objc+requires_python:objc/3 objc args
ok      7.99s project+objc+requires_python:objc/4 c++ project objc subproject
skip    3.10s project+objc+requires_python:objc/5 objfw
ok      5.04s project+objcpp+requires_python:objcpp/1 simple
ok      4.35s project+objcpp+requires_python:objcpp/2 objc++ args
skip    3.10s project+objcpp+requires_python:objcpp/3 objfw
ok      0.09s analyze:commands/copyfile.meson
ok      0.12s analyze:commands/coverage.meson
ok      0.14s analyze:commands/delete_suffix.meson
ok      0.13s analyze:commands/i18n/gettext.meson
ok      0.09s analyze:commands/i18n/itstool.meson
ok      0.14s analyze:commands/i18n/msgfmthelper.meson
ok      0.10s analyze:commands/vcs_tagger.meson
ok      0.09s analyze:lib/cmake_prelude.meson
ok      0.12s analyze:modules/_test.meson
ok      0.09s analyze:modules/gnome.meson
ok      0.09s analyze:modules/i18n.meson
ok      0.41s fmt:test_all_meson_files</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>muon</li>
    <li>muon.man</li>
    <li>muonStandalone</li>
    <li>muonStandalone.man</li>
  </ul>
</details>

nekowinston committed:
muon: only run tests in `passthru.tests`

This was done to for `muonStandalone` to have minimal dependencies.

Other edits:

- Additional cleanup to the postUnpack.

- I've added `cmdlineForMuon` flags that I've learned about since
  starting work on the `setupHook`. Without these, we get warnings about
  `fortify` requiring `-O2`, and muon defaults to `-O0` when "plain"
  isn't given.

nekowinston commented on pkgs/by-name/mu/muon/package.nix:
```diff
@@ -1,42 +1,40 @@
-  ]
-  ++ lib.optionals (!embedSamurai) [ samurai ]
-  ++ lib.optionals buildDocs [
     (python3.withPackages (ps: [ ps.pyyaml ]))
```
I hope 8d14d483bc2c18c167c3d2ea0703ee927429934a sufficiently addresses this.
According to https://nixos.org/manual/nixpkgs/stable/#var-stdenv-nativeCheckInputs all the extra packages I've added for more exhaustive testing (e.g. `nasm`) shouldn't be required to build `muonStandalone`.

nekowinston commented:
After messing around with the tests some more, it seems like the highest number of tests actually pass with
```nix
nativeCheckInputs = [
    # tests
    # required for all requires_python suite tests
    (python3.withPackages (ps: [ ps.pyyaml ]))
    # test "common/66 vcstag"
    gitMinimal
    # test "common/44 pkgconfig-gen"
    pkg-config
    # test "frameworks/6 gettext"
    gettext
  ]
  ++ lib.optionals stdenv.hostPlatform.isLinux [
    # test "frameworks/7 gnome"
    glib
    libselinux
    libsepol
    libsysprof-capture
    pcre2
    util-linuxMinimal
  ]
  ++ lib.optionals stdenv.hostPlatform.isx86_64 [
    # tests "nasm/*"
    nasm
  ];
```

Some notes about everything that's still marked as skipped:
- `common/126 generated llvm ir`:
skipped, requires llvm stdenv. runs on macOS.
- `common/147 simd`:
  skipped with `warn importing unimplemented module 'unstable-simd'`.
- `common/156 config tool variable`:
  skipped, requires `llvm`.
- `common/184 openmp`:
  skipped - I don't know enough about this library to know how to fix it, if that's even desired.
- `common/221 zlib`: 
  skipped on Linux, runs on macOS.
- `common/230 external project`
  skipped with `warn importing unimplemented module 'unstable-external_project'`
- `common/272 unity`:
  skipped, as muon doesn't support unity builds AFAICT.
- python tests:
	- `python/2 extmodule`
	- `python/3 cpython`
	- `python/4 custom target depends extmodule`
	-  `python/6 failing subproject`
	- `python/8 different python versions`
	- `python/9 extmodule limited api`
	- `python/10 extmodule limited api disabled`
skipped. even when `python3` with the respective `withPackages` is present, we get `error unimplemented`.

nekowinston committed:
muon: use two-stage build, only use needed srcs

Had to work around an issue with the fetchFromSourcehut fetcher. #441478
